### PR TITLE
fix: make better defaults for indexing, scheduling, grants

### PR DESCRIPTION
**Pull request by jgpruitt**

### CODE REVIEW

File: docs/vectorizer-api-reference.md [hunk 1]
```diff
@@ -404,10 +398,35 @@ Key points about indexing:
...
+- [ai.indexing_default](#aiindexing_default): when you do not want indexes created automatically.
 - [ai.indexing_none](#aiindexing_none): when you do not want indexes created automatically.
 - [ai.indexing_diskann](#aiindexing_diskann): configure indexing using the [DiskANN algorithm](https://github.com/timescale/pgvectorscale).
 - [ai.indexing_hnsw](#aiindexing_hnsw): configure indexing using the [Hierarchical Navigable Small World (HNSW) algorithm](https://en.wikipedia.org/wiki/Hierarchical_navigable_small_world).
 
+### ai.indexing_default
+
+You use `ai.indexing_default` to use the platform-specific default value for indexing.
+
+On Timescale Cloud, the default is `ai.indexing_diskann()`. On self-hosted, the default is `ai.indexing_none()`.
```
Comment (alejandrodnm) [inline docs/vectorizer-api-reference.md:410]: Should we explain why?

File: docs/vectorizer-api-reference.md [hunk 2]
```diff
@@ -518,10 +537,35 @@ considerations.
 
 The available functions are:
 
+- [ai.scheduling_default](#aischeduling_default): to use the plwhen you want manual control over when the vectorizer runs. Use this when you're using an external scheduling system, as is the case with self-hosted deployments.
```
Comment (alejandrodnm) [inline docs/vectorizer-api-reference.md:540]:
```suggestion
- [ai.scheduling_default](#aischeduling_default): to use when you want manual control over when the vectorizer runs. Use this when you're using an external scheduling system, as is the case with self-hosted deployments that use the pgai vectorizer worker.
```

I added the last part to tie it with the vectorizer worker. Feel free to remove that part, the first part it's a typo

File: docs/vectorizer-api-reference.md [hunk 3]
```diff
@@ -518,10 +537,35 @@ considerations.
...
 - [ai.scheduling_none](#aischeduling_none): when you want manual control over when the vectorizer runs. Use this when you're using an external scheduling system, as is the case with self-hosted deployments.
 - [ai.scheduling_timescaledb](#aischeduling_timescaledb): leverages TimescaleDB's robust job scheduling system, which is designed for reliability and scalability. Use this when you're using Timescale Cloud.
 
 
+### ai.scheduling_default
+
+You use `ai.scheduling_default` to use the platform-specific default scheduling configuration.
+
+On Timescale Cloud, the default is `ai.scheduling_timescaledb()`. On self-hosted, the default is `ai.scheduling_none()`.
```
Comment (alejandrodnm) [inline docs/vectorizer-api-reference.md:549]: Should we explain why it's different in one vs the other?

File: projects/extension/sql/idempotent/011-grant-to.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
+-------------------------------------------------------------------------------
+-- grant_to
+create or replace function ai.grant_to(variadic grantees name[]) returns name[]
+as $func$
+    select pg_catalog.array_agg(cast(x as name))
+    from (
+        select pg_catalog.unnest(grantees) x
+        union
+        select pg_catalog.unnest(pg_catalog.string_to_array(pg_catalog.current_setting('ai.grant_to_default', true), ',')) x
```
Comment (jgpruitt) [inline projects/extension/sql/idempotent/011-grant-to.sql:9]: image
Having a username with a comma in it is unlikely but possible.

File: docs/vectorizer-api-reference.md (hunk 2 above)
Comment (JamesGuthrie) [reply docs/vectorizer-api-reference.md]: This was actually a copy-paste error, thanks for the catch.

File: projects/extension/sql/idempotent/011-grant-to.sql (hunk 4 above)
Comment (jgpruitt) [reply projects/extension/sql/idempotent/011-grant-to.sql]: not worth the effort

File: docs/vectorizer-api-reference.md (hunk 3 above)
Comment (jgpruitt) [reply docs/vectorizer-api-reference.md:551]: done

**Pull request merged**